src/Kernel/Mnemonics/_if_acmpeq.php

```diff
@@ -13,10 +15,14 @@ public function execute(): void
     {
         $offset = $this->readShort();
 
+        /**
+         * @var $rightOperand _String|_Utf8
+         * @var $leftOperand _String|_Utf8
+         */
         $rightOperand = $this->popFromOperandStack();
         $leftOperand = $this->popFromOperandStack();
 
-        if ($leftOperand === $rightOperand) {
+        if ($leftOperand->hashCode() === $rightOperand->hashCode()) {
```

Comment:
Are you serious? In Java world,

```java
System.out.println("hello".hashCode() == new String("hello").hashCode());
```

prints true. If that is the case, please ignore this comment.

Reply:
No, it is a trial implementation.

Reply:
OK, thanks to have a look at this.